### PR TITLE
Rewrite stable hash computation again

### PR DESCRIPTION
**Pull request by brycemcanally**
This PR further simplifies the stable hash computation and fixes an issue with the previous implementation. The stable hash algorithm is now:
- If the file is fully contained in one file set, then we can just use the data ref hashes we store in the index for the file.
- If the file is spread across multiple file sets, then we need to run the content through a chunk writer to get the stable split points for the content (this will take advantage of the cheap copy in the common case where the file is spread across multiple file sets because it is large).

### CODE REVIEW

File: src/internal/storage/fileset/util.go [hunk 1]
```diff
@@ -144,3 +145,14 @@ func (i *Iterator) Next() (File, error) {
 		return nil, err
 	}
 }
+
+func hashDataRefs(dataRefs []*chunk.DataRef) ([]byte, error) {
+	h := pachhash.New()
+	for _, dataRef := range dataRefs {
+		_, err := h.Write(dataRef.Hash)
```
Comment (brendoncarroll) [inline src/internal/storage/fileset/util.go:152]: This is 32 bytes raw, not hex right?

File: src/internal/storage/fileset/util.go [hunk 2]
```diff
@@ -144,3 +145,14 @@ func (i *Iterator) Next() (File, error) {
 		return nil, err
 	}
 }
+
+func hashDataRefs(dataRefs []*chunk.DataRef) ([]byte, error) {
```
Comment (brendoncarroll) [inline src/internal/storage/fileset/util.go:149]: I did like having the public function `StableHash` is that gone now?

File: src/internal/storage/fileset/util.go (hunk 1 above)
Comment (brycemcanally) [reply src/internal/storage/fileset/util.go:152]: Yeah.

File: src/internal/storage/fileset/util.go (hunk 2 above)
Comment (brycemcanally) [reply src/internal/storage/fileset/util.go:149]: Yeah, it is replaced by the `Hash` function on the fileset.File interface which provides a stable hash.

**Pull request merged**